### DISCUSSION TITLE
Documentation MorphAnalysis Key Value Pair

### DISCUSSION THREAD

**Discussion by Tim-HC** [explosion/spaCy#13143] [Help: Model Advice] [answered]
Hello, I'm seeking documentation for the code snippet 
'`token.morph.to_dict()`'. 
I've identified the keys on this page :
https://github.com/explosion/spaCy/blob/master/spacy/attrs.pyx
But I'm currently unable to locate information on the potential values associated with each key. 
Any guidance on where to find this information would be greatly appreciated.

**Comment by adrianeboyd** [comment, marked as answer]
The list in `attrs.pyx` is an old legacy list, not what you want to refer to.

The individual features and values typically come from the labels in the `morphologizer` component or rules in the `attribute_ruler` component. They vary depending on which training corpus was used or what's in the hand-written `attribute_ruler` rules.

In general most of them come from the UD feats annotation: https://universaldependencies.org/u/feat/all.html
